src/app/controllers/saves_controller.rb

```diff
@@ -0,0 +1,33 @@
+class SavesController < ApplicationController
+  before_action :set_function, only: %i[create destroy]
+  before_action :set_save, only: :destroy
+
+  LIMIT = 30
+
+  def index
+    offset = params.fetch(:offset, 0).to_i
+    @saves = current_user.saves.limit(LIMIT).offset(offset).order(created_at: :desc).all
+    @next = offset + LIMIT if current_user.saves.count == LIMIT
+  end
+
+  def create
+    @save = current_user.saves.create_with(function: @function, user: current_user).new
+    @save.save
+    redirect_back(fallback_location: root_path)
```

Comment:
[R] you need to redirect back to the function not the root path, imitate the likes controller here and in the `#destroy` action also